apps/meteor/client/views/admin/moderation/ModerationConsoleTable.tsx

```diff
@@ -0,0 +1,196 @@
+
+	const query = useDebouncedValue(
+		useMemo(
+			() => ({
+				selector: text,
+				sort: JSON.stringify({ [sortBy]: sortDirection === 'asc' ? 1 : -1 }),
+				count: itemsPerPage,
+				offset: current,
+				...(end && { latest: `${new Date(end).toISOString().slice(0, 10)}T23:59:59.999Z` }),
+				...(start && { oldest: `${new Date(start).toISOString().slice(0, 10)}T00:00:00.000Z` }),
+			}),
+			[current, end, itemsPerPage, sortBy, sortDirection, start, text],
+		),
+		500,
+	);
+
+	const getReports = useEndpoint('GET', '/v1/moderation.reportsByUsers');
+
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	const {
+		data,
+		refetch: reloadReports,
+		isLoading,
+		isSuccess,
+	} = useQuery(
+		['moderation.reports', query],
+		async () => {
+			const reports = await getReports(query);
+			return reports;
+		},
+		{
+			onError: (error) => {
+				dispatchToastMessage({ type: 'error', message: error });
+			},
+		},
+	);
+
+	useEffect(() => {
+		reload.current = reloadReports;
+	}, [reload, reloadReports]);
+
+	const onChange = useMutableCallback(() => {
+		reloadReports();
+	});
+
+	const handleClick = useMutableCallback((id): void => {
+		moderationRoute.push({
+			context: 'info',
+			id,
+		});
+	});
+
+	// header sequence would be: name, reportedMessage, room, postdate, reports, actions
+	const headers = useMemo(
+		() => [
+			<GenericTableHeaderCell
+				key={'name'}
+				direction={sortDirection}
+				active={sortBy === 'reports.message.u.username'}
+				onClick={setSort}
+				sort='reports.message.u.username'
+			>
+				{t('Name')}
+			</GenericTableHeaderCell>,
+			mediaQuery && (
+				<GenericTableHeaderCell
+					w='x140'
+					key='username'
+					direction={sortDirection}
+					active={sortBy === 'reports.message.u.username'}
+					onClick={setSort}
+					sort='reports.message.u.username'
+				>
+					{t('Username')}
+				</GenericTableHeaderCell>
+			),
+			<GenericTableHeaderCell
+				key={'reportedMessage'}
+				direction={sortDirection}
+				active={sortBy === 'reports.description'}
+				onClick={setSort}
+				sort='reports.description'
+			>
+				{t('Moderation_Reported_message')}
+			</GenericTableHeaderCell>,
+			<GenericTableHeaderCell key={'room'} direction={sortDirection}>
+				{t('Room')}
+			</GenericTableHeaderCell>,
+			<GenericTableHeaderCell
+				key={'postdate'}
+				direction={sortDirection}
+				active={sortBy === 'reports.ts'}
+				onClick={setSort}
+				sort='reports.ts'
+			>
+				{t('Moderation_Report_date')}
+			</GenericTableHeaderCell>,
+			<GenericTableHeaderCell key={'reports'} direction={sortDirection} active={sortBy === 'count'} onClick={setSort} sort='count'>
+				{t('Moderation_Report_plural')}
+			</GenericTableHeaderCell>,
+			<GenericTableHeaderCell key={'actions'} width={'5%'} />,
+		],
+		[sortDirection, sortBy, setSort, t, mediaQuery],
+	);
+
+	return (
+		<>
+			<FilterByText autoFocus placeholder={'Search'} onChange={({ text }): void => setText(text)} />
```

Comment:
`search` missing translation and maybe `username` would be more useful instead

Reply:
Actually the search filter could be used to search across not only the usernames but also the reported messages, so I have added the `t('search')`. Thank you!